solidity/test/unit/UpdateStorageMirrorGuard.t.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.4 <0.9.0;
+
+// solhint-disable-next-line
+import 'forge-std/Test.sol';
+
+import {Enum} from 'safe-contracts/common/Enum.sol';
+import {UpdateStorageMirrorGuard} from 'contracts/UpdateStorageMirrorGuard.sol';
+import {IGuardCallbackModule} from 'interfaces/IGuardCallbackModule.sol';
+
+abstract contract Base is Test {
+  address public safe;
+  IGuardCallbackModule public guardCallbackModule;
+  UpdateStorageMirrorGuard public updateStorageMirrorGuard;
+
+  bytes32 public immutable SETTINGS_HASH = keccak256(abi.encodePacked('settings'));
+
+  function setUp() public {
+    safe = makeAddr('safe');
+    guardCallbackModule = IGuardCallbackModule(makeAddr('guardCallbackModule'));
+    updateStorageMirrorGuard = new UpdateStorageMirrorGuard(guardCallbackModule);
+  }
+}
+
+contract UnitUpdateStorageMirrorGuard is Base {
+  function testCheckTransaction() public {
+    assertFalse(updateStorageMirrorGuard.didSettingsChange());
+    assertEq(updateStorageMirrorGuard.settingsHash(), bytes32(''));
+
+    vm.prank(safe);
+    updateStorageMirrorGuard.checkTransaction(
+      address(0), 0, '', Enum.Operation.Call, 0, 0, 0, address(0), payable(0), '', address(0)
+    );
+
+    assertTrue(updateStorageMirrorGuard.didSettingsChange());
+    assertEq(updateStorageMirrorGuard.settingsHash(), keccak256(abi.encodePacked('settings')));
```

Comment:
In the storage mirror tests we follow a pattern of `assertEq(<assertion>, <reason>)` I think this pattern is great as it makes is really easy to debug as if a test fails we will see the reason, I think we should add these here or remove them from the storage mirror